### PR TITLE
validate sp profile secret

### LINKED ISSUE #1185: v20170131 is missing service principal profile secret validation
**Is this a request for help?**: No

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): ISSUE

---

**What version of acs-engine?**: HEAD

---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**: Kubernetes

**What happened**: v20170131 is missing service principal profile secret validation

**What you expected to happen**: missing secret should be rejected

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by weinong**
**What this PR does / why we need it**: ServicePrincipalProfile.Secret validation is missed in v20170131.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1185 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1187)

### CODE REVIEW

**Comment by weinong** [comment]
@anhowe

File: pkg/api/v20170131/validate.go [hunk 1]
```diff
@@ -100,6 +100,16 @@ func (a *Properties) Validate() error {
...
 
+	if a.OrchestratorProfile.OrchestratorType == Kubernetes {
+		if a.ServicePrincipalProfile == nil {
+			return fmt.Errorf("ServicePrincipalProfile must be specified with Orchestrator %s", a.OrchestratorProfile.OrchestratorType)
+		}
+
+		if len(a.ServicePrincipalProfile.Secret) == 0 {
+			return fmt.Errorf("service principal client secret must be specified with Orchestrator %s", a.OrchestratorProfile.OrchestratorType)
+		}
+	}
```
Comment (amanohar) [inline pkg/api/v20170131/validate.go]: Also needed in https://github.com/acs-engine/blob/master/pkg/api/v20170701/validate.go?

Comment (weinong) [inline pkg/api/v20170131/validate.go]: v20170701 has the validation.

Comment (amanohar) [inline pkg/api/v20170131/validate.go]: I meant: `if a.ServicePrincipalProfile == nil`  check in v20170701.

**Comment by seanknox** [comment]
Thanks for the tests! 👍

File: pkg/api/v20170131/validate.go (hunk 1 above)
Comment (weinong) [inline pkg/api/v20170131/validate.go]: you are right. Updated!

**Pull request merged**